burn-train/src/learner/builder.rs

```diff
@@ -33,6 +34,14 @@ where
     directory: String,
     grad_accumulation: Option<usize>,
     devices: Vec<B::Device>,
+    metric_logger_train: Option<Box<dyn MetricLogger + 'static>>,
+    metric_logger_valid: Option<Box<dyn MetricLogger + 'static>>,
+    renderer: Option<Box<dyn DashboardRenderer + 'static>>,
+    metrics_train: Vec<Box<dyn DashboardMetric<T>>>,
+    metrics_valid: Vec<Box<dyn DashboardMetric<V>>>,
+    metrics_train_numeric: Vec<Box<dyn DashboardNumericMetric<T>>>,
+    metrics_valid_numeric: Vec<Box<dyn DashboardNumericMetric<V>>>,
```

Comment:
I think we should put all metrics in a struct. It wasn't flexible enough in the dashboard, but we should have a struct that just group those types.